File: laravel_bart/SchoolApp/app/Models/Student.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Tymon\JWTAuth\Contracts\JWTSubject;
class Student extends  Authenticatable implements JWTSubject
{
    use HasFactory;
    protected $fillable = [
        'f_name',
        'm_name',
        'l_name',
        'full_name',
        'mother_name',
        'address_id',
        'birthday',
        'phone',
        'image',
        'email',
        'parent_phone',
        'class_room_id',
        'password'
        
    ];

    public function address()
    {
      return $this->belongsTo(Address::class);

    }
    public function classRoom()
    {
      return $this->belongsTo(ClassRoom::class);

    }
    public function vauationl()
    {
      return $this->hasMany(Valuation::class);

    }
   
    public function student_Note()
    {
      return $this->hasMany(Student_Note::class);

    }
    public function mark()
    {
      return $this->hasMany(Mark::class);

    }
    public function class_StudyFile()
    {
      return $this->hasMany(Class_StudyFile::class);

    }
    public function absence_Let_st()
    {
      return $this->hasMany(Absence_Let_st::class);

    }
    public $with=['address'];
       /**
     * Get the identifier that will be stored in the subject claim of the JWT.
     *
     * @return mixed
     */
    public function getJWTIdentifier()
    {
        return $this->getKey();
    }

    /**
     * Return a key value array, containing any custom claims to be added to the JWT.
     *
     * @return array
     */
    public function getJWTCustomClaims()
    {
        return [];
    }

    protected $hidden = [
      'password',
      'remember_token',
  ];
}

File: flutter_bart/tr_app/lib/Screen/login/Models/student.dart
// To parse this JSON data, do
//
//     final studentModel = studentModelFromJson(jsonString);

import 'dart:convert';

StudentModel studentModelFromJson(String str) =>
    StudentModel.fromJson(json.decode(str));

String studentModelToJson(StudentModel data) => json.encode(data.toJson());

class StudentModel {
  StudentModel({
    required this.id,
    required this.fName,
    required this.mName,
    required this.lName,
    required this.fullName,
    required this.motherName,
    required this.image,
    required this.addressId,
    required this.birthday,
    required this.phone,
    required this.email,
    required this.parentPhone,
    required this.classRoomId,
  });

  int id;
  String fName;
  String mName;
  String lName;
  String fullName;
  String motherName;
  String image;
  int addressId;
  String birthday;
  String phone;
  String email;
  String parentPhone;
  int classRoomId;

  factory StudentModel.fromJson(Map<String, dynamic> json) => StudentModel(
        id: json["id"],
        fName: json["f_name"],
        mName: json["m_name"],
        lName: json["l_name"],
        fullName: json["full_name"],
        motherName: json["mother_name"],
        image: json["image"],
        addressId: json["address_id"],
        birthday: json["birthday"],
        phone: json["phone"],
        email: json["email"],
        parentPhone: json["parent_phone"],
        classRoomId: json["class_room_id"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "f_name": fName,
        "m_name": mName,
        "l_name": lName,
        "full_name": fullName,
        "mother_name": motherName,
        "image": image,
        "address_id": addressId,
        "birthday": birthday,
        "phone": phone,
        "email": email,
        "parent_phone": parentPhone,
        "class_room_id": classRoomId,
      };
}
